### PR TITLE
fix: move nss jobs into plugin, update docs

### PR DESCRIPTION
**Pull request by mckornfield**
## Summary by CodeRabbit

* **New Features**
  * Added Safe Synthesizer SDK: resources (sync/async jobs), SafeSynthesizerJob wrapper (polling/status/logs/data/summary/report/Notebook display), ReportHtml, and a fluent SafeSynthesizerJobBuilder for building/submitting jobs and reusing pretrained jobs.
* **Bug Fixes**
  * Fixed pretrained-model config handling at runtime and tightened local-run env isolation to avoid leaking parent context.
* **Documentation**
  * Revised tutorials, CLI/README, workflows, and site nav to clarify local vs platform flows and document SDK.
* **Tests / Chores**
  * Added unit tests, packaging entry points, and third‑party license entries.

### CODE REVIEW

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/config.py
Comment (mckornfield) [inline plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/config.py:1]: these SDK files are just a port of something I deleted in the old place

File: docs/safe-synthesizer/tutorials/safe-synthesizer-101.md [hunk 1]
```diff
@@ -164,7 +175,6 @@ builder = (
     SafeSynthesizerJobBuilder(client)
     .with_data_source(df)
     .with_classify_model_provider(provider_name)  # Enable column classification
-    .with_replace_pii()  # Enable PII replacement
```
Comment (anastasia-nesterenko) [inline docs/safe-synthesizer/tutorials/safe-synthesizer-101.md:167]: nit: with with_replace_pii() deleted, should the section `**What happens next:**` be modified?

File: docs/safe-synthesizer/about/jobs.md
Comment (anastasia-nesterenko) [inline docs/safe-synthesizer/about/jobs.md]: nit: old name usage

File: docs/safe-synthesizer/about/reference.md
Comment (anastasia-nesterenko) [inline docs/safe-synthesizer/about/reference.md]: nit: same old name usage

**Review by anastasia-nesterenko — APPROVED** [review]
left some nit-picks

File: docs/safe-synthesizer/tutorials/safe-synthesizer-101.md [hunk 2]
```diff
@@ -164,7 +175,6 @@ builder = (
     SafeSynthesizerJobBuilder(client)
     .with_data_source(df)
     .with_classify_model_provider(provider_name)  # Enable column classification
```
Comment (tylersbray) [inline docs/safe-synthesizer/tutorials/safe-synthesizer-101.md:177]: > **agent** (`pr-review`): Tutorial regression — classify provider never reaches job spec

**Human spot-check:** Step 6 verifies a classify model provider and Step 7 calls `.with_classify_model_provider()`, but `.with_replace_pii()` was removed from this builder chain. Users follow classification setup that the submitted job ignores.

**Root cause:** `SafeSynthesizerJobBuilder._build_job_spec()` only injects `classify_model_provider` when `enable_replace_pii` is true (`job_builder.py` L198–203). The API compiler reads it only from `config.replace_pii.globals.classify`.

**Agent task — pick one:**
- **Option A (restore prior behavior):** Add `.with_replace_pii()` before `.synthesize()` on this chain.
- **Option B (synthesis-only tutorial):** Remove/reword Step 6 and drop `.with_classify_model_provider()` here.

**Acceptance criteria:**
```python
# Option A: spec must include
config["enable_replace_pii"] is True
config["replace_pii"]["globals"]["classify"]["classify_model_provider"] == provider_name
# Option B: spec must NOT reference classify_model_provider in the main job
```

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job_builder.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        self._time_series_config = _merge_config(config, kwargs)
+        return self
+
+    def with_replace_pii(self, config: _ConfigInput = None, **kwargs: Any) -> Self:
+        """Configure and enable PII replacement."""
+        self._replace_pii_config = _merge_config(config, kwargs)
+        self._enable_replace_pii = True
+        return self
+
+    def with_classify_model_provider(self, provider_name: str) -> Self:
```
Comment (tylersbray) [inline plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job_builder.py:115]: > **agent** (`pr-review`): `with_classify_model_provider()` is a no-op without PII replacement

**Human spot-check:** Callers can chain `.with_classify_model_provider("default/nvidia-build")` on a synthesis-only job and believe classification is enabled. It is silently ignored unless `.with_replace_pii()` was also called.

**Related:** Same issue drives the `safe-synthesizer-101.md` tutorial regression.

**Agent task — pick one:**
1. **Doc-only:** Add a docstring/note on `with_classify_model_provider()` stating it requires `.with_replace_pii()`.
2. **API fix:** If classification should work independently of PII replacement, inject `classify_model_provider` into a synthesis-accessible config path (and update `endpoints.py` compiler). Confirm product intent first.

**Acceptance criteria:** Either explicit docs warning OR spec contains classify config without `enable_replace_pii`.

File: docs/_snippets/nvidia-build-model-provider.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 !!! note
-    The platform pre-configures a `system/nvidia-build` model provider during startup.
+    The platform pre-configures a `default/nvidia-build` model provider during startup.
```
Comment (tylersbray) [inline docs/_snippets/nvidia-build-model-provider.md]: > **agent** (`pr-review`): `default/nvidia-build` may be wrong for platform-seed deployments

**Human spot-check:** This shared snippet now says `default/nvidia-build`. That matches `nemo setup` (`_auto_setup(client, "default")`), but **platform-seed** still creates the provider in the `system` workspace (`services/platform-seed/src/nmp/platform_seed/tasks/seed/run.py`). Helm/K8s users may not find `default/nvidia-build`.

**Agent task — pick one:**
1. Document both paths: `nemo setup` → `default/nvidia-build`; platform-seed → `system/nvidia-build`.
2. Align platform-seed to `default` workspace (bigger change, out of scope?).
3. De-emphasize hard-coded workspace: tell users to run `nemo inference providers list` and use whatever workspace/name exists.

**Acceptance criteria:** Snippet is accurate for both local setup and seeded deployments, or explicitly scopes itself to `nemo setup` users only.

**Review by tylersbray — COMMENTED** [review]
> **agent** (`pr-review`): Full review summary

Solid PR overall — plugin SDK relocation, `_runtime_job_config`, and scheduler workspace fix are the right moves. Inline comments below are ordered roughly by severity.

**Merge blockers / high priority**
1. `safe-synthesizer-101.md` — classify provider configured but never submitted (missing `.with_replace_pii()`)
2. `nvidia-build-model-provider.md` — `default/` vs `system/` workspace mismatch with platform-seed

**Should fix in this PR**
3. Stale "Host-Local Development and Testing" link titles in `jobs.md`, `reference.md`, `about/index.md`
4. Add `docs/safe-synthesizer/sdk-resources.md` (parity with anonymizer/evaluator/data-designer)
5. `AsyncSafeSynthesizerJobsResource.get_logs` should be `async def`
6. Expand tests for `SafeSynthesizerJob` and `_runtime_job_config` edge cases

**Confirm intent / lower priority**
7. `mkdocs.yml` nav pruning — jobs/reference/DP tutorial removed from published nav
8. `job_builder.py` — `with_classify_model_provider()` only applies when PII replacement enabled
9. `job.py` — broad `except Exception` in `wait_for_completion`
10. `config.py` — silent `ImportError` leaves empty `__all__`
11. `packages/nemo_platform/BUNDLING.md` still references removed `safe_synthesizer_sdk` vendoring (file not in this PR)

**Looks good**
- `enabled-plugins` addition, SDK entry points, `test_sdk.py` resource/builder coverage, scheduler workspace test

File: docs/safe-synthesizer/about/jobs.md [hunk 5]
```diff
@@ -120,7 +120,7 @@ import os
 import pandas as pd
 
 from nemo_platform import NeMoPlatform
-from nemo_platform.beta.safe_synthesizer.job_builder import SafeSynthesizerJobBuilder
+from nemo_safe_synthesizer_plugin.sdk.job_builder import SafeSynthesizerJobBuilder
```
Comment (tylersbray) [inline docs/safe-synthesizer/about/jobs.md:123]: > **agent** (`pr-review`): Stale link titles in this file

**Human spot-check:** Import path update here is correct. Same file still uses the old page title in two places:
- L112: `[Host-Local Development and Testing](host-local-development.md)`
- L312: same label in Related topics

Nav/getting-started renamed this page to **"Local and Subprocess Execution"**.

**Agent task:**
```markdown
# Replace both occurrences:
[Local and Subprocess Execution](host-local-development.md)
```

**Also update** `docs/safe-synthesizer/about/reference.md` L20 and `docs/safe-synthesizer/about/index.md` L109 (not touched in this PR).

File: docs/safe-synthesizer/about/reference.md [hunk 6]
```diff
@@ -95,7 +95,7 @@ import os
 import pandas as pd
 
 from nemo_platform import NeMoPlatform
-from nemo_platform.beta.safe_synthesizer.job_builder import SafeSynthesizerJobBuilder
+from nemo_safe_synthesizer_plugin.sdk.job_builder import SafeSynthesizerJobBuilder
```
Comment (tylersbray) [inline docs/safe-synthesizer/about/reference.md:98]: > **agent** (`pr-review`): Stale link title on L20

**Human spot-check:** Import path update (this hunk) is correct. L20 still says:
`[Host-Local Development and Testing](host-local-development.md)`

**Agent task:**
```markdown
[Local and Subprocess Execution](host-local-development.md)
```

Mirror the same fix in `jobs.md` L112/L312 and `about/index.md` L109.

File: docs/safe-synthesizer/getting-started.md [hunk 7]
```diff
@@ -31,15 +31,15 @@ uv run nemo safe-synthesizer run-local \
...
 ---
 
 ## Next Steps
 
-Run one of the [tutorials](tutorials/index.md) to create your first synthetic dataset:
+Create your first synthetic dataset:
 
-- [Safe Synthesizer 101 Tutorial](tutorials/safe-synthesizer-101.md) - A beginner-friendly introduction
-- [Differential Privacy Tutorial](tutorials/differential-privacy.md) - Generate differentially-private synthetic data
+- [Safe Synthesizer 101 Tutorial](tutorials/safe-synthesizer-101.md) - a beginner-friendly introduction
```
Comment (tylersbray) [inline docs/safe-synthesizer/getting-started.md:42]: > **agent** (`pr-review`): Missing SDK resources doc page

**Human spot-check:** Anonymizer, Evaluator, and Data Designer each have a `docs/<plugin>/sdk-resources.md` page linked from nav. This PR adds substantial SDK surface (`client.safe_synthesizer.jobs`, `SafeSynthesizerJobBuilder`, `SafeSynthesizerJob`) but no equivalent doc.

**Agent task:**
1. Create `docs/safe-synthesizer/sdk-resources.md` modeled on `docs/anonymizer/sdk-resources.md`.
2. Cover: `NeMoPlatform.safe_synthesizer`, `SafeSynthesizerJobsResource` (create/list/retrieve), `SafeSynthesizerJobBuilder`, `SafeSynthesizerJob` (wait/logs/results).
3. Add nav entry under `Synthesize Safe Data` in `mkdocs.yml`.
4. Link from `getting-started.md` Next Steps.

**Acceptance criteria:** Published docs show import paths and method table without requiring the 101 tutorial.

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/resources.py [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+            headers=http_utils.platform_default_headers(self._platform),
+        )
+        _raise_for_status(response)
+        return _object_from_mapping(response.json())
+
+    async def get_status(self, name: str, *, workspace: str | None = None) -> Any:
+        """Retrieve Safe Synthesizer job status."""
+        return await self._platform.jobs.get_status(name, workspace=workspace)
+
+    def get_logs(self, name: str, *, workspace: str | None = None, **kwargs: Any) -> Any:
```
Comment (tylersbray) [inline plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/resources.py]: > **agent** (`pr-review`): `AsyncSafeSynthesizerJobsResource.get_logs` should be async

**Human spot-check:** `create`, `list`, `retrieve`, and `get_status` on the async resource are `async def`. `get_logs` (L166) is a plain `def` that calls the sync `self._platform.jobs.get_logs`.

**Agent task:**
```python
async def get_logs(self, name: str, *, workspace: str | None = None, **kwargs: Any) -> Any:
    return await self._platform.jobs.get_logs(name, workspace=workspace, **kwargs)
```
(Confirm `AsyncNeMoPlatform.jobs.get_logs` is awaitable.)

**Acceptance criteria:** Async resource surface is consistently `async def` for all public methods.

File: plugins/nemo-safe-synthesizer/tests/unit/test_sdk.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        .with_generate(num_records=25)
+    )
+
+    job = builder.create_job(name="adapter-reuse-job")
+
+    assert job.job_name == "adapter-reuse-job"
+    create_kwargs = client.safe_synthesizer.jobs.create.call_args.kwargs
+    assert create_kwargs["spec"]["pretrained_model_job"] == "first-synth-job"
+    assert create_kwargs["spec"]["config"]["generation"] == {"num_records": 25}
+    assert "pretrained_model" not in create_kwargs["spec"]["config"]["training"]
```
Comment (tylersbray) [inline plugins/nemo-safe-synthesizer/tests/unit/test_sdk.py:147]: > **agent** (`pr-review`): Add tests for `SafeSynthesizerJob` high-level wrapper

**Human spot-check:** Current tests cover resource HTTP routing and `SafeSynthesizerJobBuilder` spec assembly. No coverage for `sdk/job.py` (`SafeSynthesizerJob`).

**Agent task — add tests with mocked `NeMoPlatform`:**
1. `wait_for_completion` raises `RuntimeError` on `error`/`cancelled` status.
2. `fetch_data` reads CSV from `client.jobs.results.download("synthetic-data", ...)`.
3. `fetch_summary` parses JSON into `SafeSynthesizerSummary`.
4. (Optional) `fetch_logs` pagination follows `next_page`.

**Acceptance criteria:** `uv run pytest plugins/nemo-safe-synthesizer/tests/unit/test_sdk.py -v` covers `SafeSynthesizerJob` without network.

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/api/v2/jobs/endpoints.py [hunk 10]
```diff
@@ -154,6 +154,15 @@ def parse_pretrained_model_job_ref(job_ref: str, workspace_fallback: str) -> tup
     return workspace, job_name
 
 
+def _runtime_job_config(job_config: SafeSynthesizerJobConfig) -> dict[str, Any]:
```
Comment (tylersbray) [inline plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/api/v2/jobs/endpoints.py:46]: > **agent** (`pr-review`): Add edge-case tests for `_runtime_job_config`

**Human spot-check:** `_runtime_job_config` correctly strips `training.pretrained_model` when `pretrained_model_job` is set — good fix. `test_plugin_job_config_allows_pretrained_model_job_runtime_config` covers the happy path.

**Agent task — add unit tests in `test_jobs.py`:**
1. `pretrained_model_job` set but `config.training` missing → no error, config passes through.
2. `pretrained_model_job` set but `config.training` is not a dict → no error.
3. `pretrained_model_job` unset → `training.pretrained_model` preserved.

**Acceptance criteria:** `_runtime_job_config` behavior is locked for non-dict/missing `training` shapes.

File: mkdocs.yml [hunk 11]
```diff
@@ -320,19 +326,9 @@ nav:
     - SDK Resources: data-designer/sdk-resources.md
     - Migrating from Standalone Library: data-designer/migration.md
   - Synthesize Safe Data:
```
Comment (tylersbray) [inline mkdocs.yml:320]: > **agent** (`pr-review`): Confirm intentional nav pruning

**Human spot-check:** Published `Synthesize Safe Data` nav shrinks from full About + tutorials to only:
- Getting Started
- Local and Subprocess Execution
- Safe Synthesizer 101

Removed from nav (moved to `hidden_docs`): jobs.md, reference.md, differential-privacy.md, about/index.md, evaluation, PII replacement, etc.

**Question for author:** Is this deliberate scope reduction for the public docs site?

**Agent task if intentional:**
- Add cross-links from Getting Started / 101 to hidden pages for readers using `make live-with-unready`.
- Consider keeping Differential Privacy or Jobs in nav if still user-facing.

**Agent task if unintentional:** Restore About subsection and DP tutorial to nav.

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job.py [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+                    httpx_logger.setLevel("ERROR")
+                    new_logs, last_page_cursor = self._fetch_logs_incremental(
+                        page_cursor=last_page_cursor, timeout=log_timeout
+                    )
+                    for new_log in new_logs:
+                        log_key = f"{new_log.timestamp}:{hash(new_log.message)}"
+                        if log_key not in seen_log_keys:
+                            print(new_log.message.strip())
+                            seen_log_keys.add(log_key)
+                except Exception:
```
Comment (tylersbray) [inline plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job.py]: > **agent** (`pr-review`): Narrow exception handling in `wait_for_completion`

**Human spot-check:** Log polling uses `except Exception:` which swallows all failures during status wait (network blips OK, but also programming errors).

**Agent task — pick one:**
1. Catch `httpx.HTTPError` (and maybe `TimeoutException`) only; re-raise others.
2. Keep broad catch but log at warning without `logger.exception` (full traceback) unless `verbose` and repeated failures.

**Acceptance criteria:** Transient log-fetch failures don't abort wait; unexpected exceptions still surface.

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/config.py [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        "DifferentialPrivacyHyperparams",
+        "EvaluationParameters",
+        "GenerateParameters",
+        "PiiReplacerConfig",
+        "SafeSynthesizerJobConfig",
+        "SafeSynthesizerParameters",
+        "TimeSeriesParameters",
+        "TrainingHyperparams",
+    ]
+except ImportError:
```
Comment (tylersbray) [inline plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/config.py]: > **agent** (`pr-review`): Silent `ImportError` hides missing dependency

**Human spot-check:** If `nemo_safe_synthesizer` is not installed, `config.py` sets `__all__ = []` with no warning. Users get confusing `ImportError` from deeper imports.

**Agent task — pick one:**
1. Re-raise `ImportError` with message: "Install nemo-safe-synthesizer to use SDK config types."
2. Emit `logger.warning(...)` once in the `except` block.

**Acceptance criteria:** Missing optional dep produces actionable error, not empty module.

File: pyproject.toml [hunk 14]
```diff
@@ -168,6 +168,7 @@ enabled-plugins = [
   "nemo-evaluator-plugin",
   "nemo-guardrails-plugin",
   "nemo-auditor-plugin",
+  "nemo-safe-synthesizer-plugin",
```
Comment (tylersbray) [inline pyproject.toml:171]: > **agent** (`pr-review`): Good — required for SDK mount

**Human spot-check:** Adding `nemo-safe-synthesizer-plugin` to `enabled-plugins` is necessary so `discover_sdk()` exposes `client.safe_synthesizer` after a normal `uv sync` / bootstrap. Without this, tutorial/SDK examples fail unless users manually enable the plugin.

No change needed — flagging as intentional and correct.

**Review by tylersbray — APPROVED** [review]
Left a few comments...

File: docs/safe-synthesizer/tutorials/safe-synthesizer-101.md (hunk 1 above)
Comment (mckornfield) [reply docs/safe-synthesizer/tutorials/safe-synthesizer-101.md:167]: yeah... I'll put this back (removed it to speed up testing)

File: docs/safe-synthesizer/about/jobs.md
Comment (mckornfield) [reply docs/safe-synthesizer/about/jobs.md]: nooooo

File: docs/safe-synthesizer/tutorials/safe-synthesizer-101.md (hunk 2 above)
Comment (mckornfield) [reply docs/safe-synthesizer/tutorials/safe-synthesizer-101.md:177]: yeah I'll put it back

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job_builder.py (hunk 3 above)
Comment (mckornfield) [reply plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job_builder.py:115]: with_replace_pii coming in hot

File: docs/_snippets/nvidia-build-model-provider.md (hunk 4 above)
Comment (mckornfield) [reply docs/_snippets/nvidia-build-model-provider.md]: this confuses the frick out of me lol. I think I want to just do the nemo setup version lol

File: docs/safe-synthesizer/about/jobs.md (hunk 5 above)
Comment (mckornfield) [reply docs/safe-synthesizer/about/jobs.md:123]: ah yeah I need to sweep these

File: docs/safe-synthesizer/getting-started.md (hunk 7 above)
Comment (mckornfield) [reply docs/safe-synthesizer/getting-started.md:42]: yeah makes sense. I'll add more back in, this PR put the tiniest amount so I don't get rekt in QA lol

File: mkdocs.yml (hunk 11 above)
Comment (mckornfield) [reply mkdocs.yml:320]: yeah I'll put MOAR in

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job.py (hunk 12 above)
Comment (mckornfield) [reply plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/job.py]: ugh I remember this could hit weird timeouts and make the logs look garbage, but might be worth cleaning up

File: plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/config.py (hunk 13 above)
Comment (mckornfield) [reply plugins/nemo-safe-synthesizer/src/nemo_safe_synthesizer_plugin/sdk/config.py]: yeah I don't know how useful this guard is personally
